tests/scenarios/subscriptions.spec.ts

```diff
@@ -924,5 +924,289 @@ export function testSubscriptionScenarios(): void {
+        //    alice checks that the correct messages were omitted
+
+        const alice = await DidKeyResolver.generate();
+        const bob = await DidKeyResolver.generate();
+        const carol = await DidKeyResolver.generate();
+
+        // create protocol
+        const protocolConfigure = await TestDataGenerator.generateProtocolsConfigure({
+          author             : alice,
+          protocolDefinition : { ...threadProtocol }
+        });
+        const protocolConfigureReply = await dwn.processMessage(alice.did, protocolConfigure.message);
+        expect(protocolConfigureReply.status.code).to.equal(202);
+        const protocol = protocolConfigure.message.descriptor.definition.protocol;
+
+        // alice creates thread
+        const thread = await TestDataGenerator.generateRecordsWrite({
+          author       : alice,
+          protocol     : protocol,
+          protocolPath : 'thread'
+        });
+        const threadReply = await dwn.processMessage(alice.did, thread.message, { dataStream: thread.dataStream });
+        expect(threadReply.status.code).to.equal(202);
+
+
+        // subscribe to this thread's events
+        const messages:string[] = [];
+        const subscriptionHandler = async (message:GenericMessage):Promise<void> => {
+          messages.push(await Message.getCid(message));
+        };
+
+        const threadSubscription = await TestDataGenerator.generateRecordsSubscribe({
+          author : alice,
+          filter : { protocol: protocol, protocolPath: 'thread', parentId: thread.message.recordId }, // thread updates
+        });
+        const threadSubscriptionReply = await dwn.processMessage(alice.did, threadSubscription.message, {
+          subscriptionHandler
+        });
+        expect(threadSubscriptionReply.status.code).to.equal(200);
+        expect(threadSubscriptionReply.subscription).to.exist;
+
+        const participantSubscription = await TestDataGenerator.generateRecordsSubscribe({
+          author : alice,
+          filter : { protocol: protocol, protocolPath: 'thread/participant', parentId: thread.message.recordId }, // participant updates
+        });
+        const participantSubscriptionReply = await dwn.processMessage(alice.did, participantSubscription.message, {
+          subscriptionHandler
+        });
+        expect(participantSubscriptionReply.status.code).to.equal(200);
+        expect(participantSubscriptionReply.subscription).to.exist;
+
+        const chatSubscription = await TestDataGenerator.generateRecordsSubscribe({
+          author : alice,
+          filter : { protocol: protocol, protocolPath: 'thread/chat', parentId: thread.message.recordId } // chat updates
+        });
+        const chatSubscriptionReply = await dwn.processMessage(alice.did, chatSubscription.message, {
+          subscriptionHandler
+        });
+        expect(chatSubscriptionReply.status.code).to.equal(200);
+        expect(chatSubscriptionReply.subscription).to.exist;
```

Comment:
For future, but this definitely feels super forced, this probably can go away if we support one-of filter for protocol path (or for all strings while at it), we also would probably one to support the same for permission grants.

Reply:
Yup was thinking the same, best for a separate effort but it shouldn't be anything crazy.